sim/firesim-lib/src/main/scala/bridges/TestFinisherBridge.scala

```diff
@@ -0,0 +1,97 @@
+
+package firesim.bridges
+
+import chisel3._
+import chisel3.util._
+import freechips.rocketchip.config.Parameters
+import midas.widgets._
+import midas.widgets.PeekPokeBridge
+
+case class CodeElaborator(isErr: Boolean, message : String)
+
+case class TestIndicatorParams(
+  nConcurrentTests: Int     = 1,
+  codeMessage: Seq[CodeElaborator]
+) {
+  def codeBits: Int    = 16
+}                 
+
+
+class TestFinisherPortIO(val params: TestIndicatorParams) extends Bundle {
+  //Error code in case program terminates unsuccessfully
+  val errCode = UInt((params.codeBits).W)
+  val valid = Bool() 
+}
+
+class TestFinisherBridgeTargetIO(params: TestIndicatorParams) extends Bundle {
+  val clock = Input(Clock())
+  val testDone = Input(new TestFinisherPortIO(params))
+}
+
+class TestFinisherBridgeHostIO(params: TestIndicatorParams)
+                              (private val targetIO : TestFinisherBridgeTargetIO = new TestFinisherBridgeTargetIO(params)) 
+                               extends Bundle with ChannelizedHostPortIO{
+  def targetClockRef = targetIO.clock 
+  //It is just inputs from the target to indicate test completion
+  //The HostPort annotation would not work with aggregates so creating individual channels.
+  val errCodeH = InputChannel(targetIO.testDone.errCode)
+  val validH = InputChannel(targetIO.testDone.valid)
+}
+
+class TestFinisherBridge(params: TestIndicatorParams)(implicit p: Parameters) extends BlackBox
+    with Bridge[TestFinisherBridgeHostIO, TestFinisherBridgeModule] {
+  val io = IO(new TestFinisherBridgeTargetIO(params))
+  val bridgeIO = new TestFinisherBridgeHostIO(params)(io)
+
+  val constructorArg = Some(params)
+
+  generateAnnotations()
+}
+
+object TestFinisherBridge {
+  def apply(clock: Clock, finisher: TestFinisherPortIO)(implicit p: Parameters): TestFinisherBridge = {
+    val tfm = Module(new TestFinisherBridge(finisher.params))
+    tfm.io.testDone := finisher
+    tfm.io.clock := clock 
+    tfm  
+  }
+}
+
+class TestFinisherBridgeModule(params: TestIndicatorParams)(implicit p: Parameters) extends BridgeModule[TestFinisherBridgeHostIO]()(p) {
+  lazy val module = new BridgeModuleImp(this) {
+    // This creates the interfaces for all of the host-side transport
+    // AXI4-lite for the simulation control bus, =
+    // AXI4 for DMA
+    val io = IO(new WidgetIO())
+
+
+
+    // This creates the host-side interface of your TargetIO
+    val hPort = IO(new TestFinisherBridgeHostIO(params)())
+    
+    val statusDone = RegInit(Bool(), false.B)
```

Comment:
You should be able to write this as: 
```suggestion
    val statusDone = RegInit(false.B)
```